### DISCUSSION TITLE
一点小疑问

### DISCUSSION THREAD

**Discussion by LBYPatrick** [shadowsocks/shadowsocks-android#2690] [Q&A]
### Environment
Android Version:                7.1.2
Device:                               Nexus 5X
Shadowsocks version:        4.1.7

我最近要回国了，自己在日本Linode搭了个SS（用shadowsocks-libev），但是我不知道VPN mode下会不会也会出那个WiFi感叹号？

**Comment by whour** [comment]
>  (以下办法支持安卓7.1.2/7.1.1)
adb shell settings put global captive_portal_https_url https://www.google.cn/generate_204
然后开启飞行模式，然后关闭飞行模式解决！亲自测试有效！
这个是Android系统级别的，修改认证服务器，和Shadowsocks没关系。

**Comment by madeye** [comment]
You can try the approach from @whour.

**Comment by LBYPatrick** [comment]
@whour 感谢！那请问如果我想改回来的话又怎么操作呢？还有，如果我不改这个的话会不会导致谷歌一直尝试连接导致耗电呢？

**Comment by LBYPatrick** [comment]
@madeye Thanks. BTW, why did I receive 2.6.4 for shadowsocks-libev instead of 3.x in 17.04 with APT repo? Is it still unstable?

**Comment by whour** [comment]
>  关闭网络验证：adb shell "settings put global captive_portal_detection_enabled 0" 
或更改网络验证服务器：
用v2ex的：adb shell "settings put global captive_portal_https_url https://captive.v2ex.co/generate_204"
或者google.cn：adb shell "settings put global captive_portal_server https://www.google.cn/generate_204"

@LBYPatrick 是否耗电这个不清楚，强迫症表示要干掉叹号。
如果想用最新版本的shadowsocks-libev 可以查看Readme来自己编译，如果编译遇到问题，搜一下Issue，一般能找到答案。

**Comment by LBYPatrick** [comment]
@whour 感谢。关于shadowsocks-libev，我在想3.0是不是个大版本更新...会不会改了相当多的东西...

**Comment by LBYPatrick** [comment]
@whour 干掉感叹号的话，我觉得还是直接关闭网络验证好了，因为我不知道默认的服务器是哪个...

**Comment by whour** [comment]
@LBYPatrick 3.0加入了AEAD算法，支持obfs混淆和kcptun等扩展，大版本更新。

**Comment by andronoob** [comment]
Maybe com.android.captiveportallogin should be whitelisted by default. I have been doing this for a while. Otherwise I won't be able to use public WiFi hotspots with captive portal.
